qutebrowser/browser/webengine/webenginesettings.py

```diff
@@ -162,6 +182,32 @@ class WebEngineSettings(websettings.AbstractSettings):
         QWebEngineSettings.FantasyFont: QFont.Fantasy,
     }
 
+    def set_unknown_url_scheme_policy(self, policy: str) -> bool:
+        """Set the UnknownUrlSchemePolicy to use.
+
+        Return:
+            True if there was a change, False otherwise.
+        """
+        old_value = self._settings.unknownUrlSchemePolicy()
+        if policy is usertypes.UNSET:  # type: ignore
```

Comment:
This shouldn't be ignored anymore, because now, `usertypes.UNSET` can be a valid, expected value for this function. Instead, the type hint should probably be changed to `typing.Union[str, usertypes.Unset]` (`UNSET` is an instance of the class `Unset`).

Reply:
Unforturnately `tox -e mypy` complainted,

> qutebrowser/browser/webengine/webenginesettings.py:192: error: invalid syntax  [syntax]
            if policy is usertypes.UNSET:  # type: typing.Union[str, usertypes.Unset]

For _"error: Non-overlapping identity check (left operand type: "str", right operand type: "Unset")_, the only solution I can find is `type: ignore`. Or shall we remove type checking for set_unknown_url_scheme_policy as like other functions in qutebrowser/browser/webengine/webenginesettings.py?

Reply:
Sorry for not being clearer - I mean the type hint for the `policy` argument, as `policy` now can be either `UNSET` or a string.

Reply:
I'm not familiar with mypy yet. Thank you for the clarification!
Is the following change OK?
```diff
diff --git a/qutebrowser/browser/webengine/webenginesettings.py b/qutebrowser/browser/webengine/webenginesettings.py
index f06e71653..00f501caf 100644
--- a/qutebrowser/browser/webengine/webenginesettings.py
+++ b/qutebrowser/browser/webengine/webenginesettings.py
@@ -182,18 +182,21 @@ class WebEngineSettings(websettings.AbstractSettings):
         QWebEngineSettings.FantasyFont: QFont.Fantasy,
     }
 
-    def set_unknown_url_scheme_policy(self, policy: str) -> bool:
+    def set_unknown_url_scheme_policy(
+                self,
+                policy: typing.Union[str, usertypes.Unset]
+    ) -> bool:
         """Set the UnknownUrlSchemePolicy to use.
 
         Return:
             True if there was a change, False otherwise.
         """
         old_value = self._settings.unknownUrlSchemePolicy()
-        if policy is usertypes.UNSET:  # type: ignore
+        if policy is usertypes.UNSET:
             self._settings.resetUnknownUrlSchemePolicy()
             new_value = self._settings.unknownUrlSchemePolicy()
         else:
-            new_value = self._UNKNOWN_URL_SCHEME_POLICY[policy]
+            new_value = self._UNKNOWN_URL_SCHEME_POLICY[policy]  # type: ignore
             self._settings.setUnknownUrlSchemePolicy(new_value)
         return old_value != new_value
 
```

Reply:
I think if you replace `if policy is usertypes.UNSET` with `if isinstance(policy, usertypes.Unset)`, mypy should understand the code flow and you can drop the second `# type: ignore` as well.

Reply:
Thank you! That is a subtle change although I expected mypy to be smarter.

Reply:
> That is a subtle change although I expected mypy to be smarter.

I'm not surprised it doesn't catch that - it doesn't have a way to know that `UNSET` is the only instance of `Unset`, i.e. that `... is UNSET` is equivalent to `isinstance(..., Unset)`. There *might* be a way for us to define `UNSET` in a way that it can know about this, but I'm not sure. I'll add some ideas to #5259.